salt/utils/schema.py

```diff
@@ -515,7 +515,7 @@ def __call__(cls, *args, **kwargs):
                 'Please pass all arguments as named arguments. Un-named '
                 'arguments are not supported'
             )
-        for key in kwargs.keys():
+        for key in kwargs.copy().keys():
```

Comment:
Could you clarify the need for this copy, please? It seems like all we should be doing here is iterating over keys.